File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int total = 0;
        int p = 0;
        int n = nums.length;
        
        for (int i = 0; i < n; i++) {
            int rem = k;
            p = i;
            while (p < n) {
                rem = rem - nums[p];
                if (rem == 0) total++;
                p++;
            }
        }
        
        return total;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        c, sv, md, n = 0, 0, {0:1}, len(nums)
        for i in range(n):
          sv += nums[i]
          if sv - k in md: c += md[sv - k]
          md[sv] = md.get(sv,0) + 1
        return c